avahi-python/avahi/__init__.py

```diff
@@ -110,3 +110,22 @@ def dict_to_txt_array(txt_dict):
         l.append(string_to_byte_array("%s=%s" % (k,v)))
 
     return l
+
+
+def txt_array_to_dict(self, txt_array):
+    txt_dict = {}
+    for els in txt_array:
+        key, val = '', None
+        for c in els:
+            c = chr(c)
```

Comment:
In Python 3 this would have to be something like `bytes((c,))` to build up a bytes object: TXT records aren't guaranteed to be in any particular encoding, so (Unicode) strings aren't a suitable rerpresentation.

Reply:
do you know any polyglot version?  I assume this module to be used by a lot of python2 projects. In fact, Debian hasn't managed yet to ship this trivially compatible module for python3. Yet, it'd be nice to have something that works on both Pythons, I guess.

Reply:
The best I can come up with is:

```
def _bytechr(c):
    """
    if str is bytes:
        return chr(c)
    else:
        return bytes((c,))
```

and use that instead of `chr`.

I think this function really needs some unit tests that can be run on both Python 2 and Python 3, with assertions like:

```
txt_array_to_dict(self, [map(ord, 'foo')]) == {b'foo': b''}
txt_array_to_dict(self, [map(ord, 'foo=')]) == {b'foo': b''}
txt_array_to_dict(self, [map(ord, 'foo=bar')]) == {b'foo': b'bar'}
txt_array_to_dict(self, [[0, 1, 2, 0xff, 0x99]]) == {b'\x00\x01\x02\xff\x99': b''}
```

Happily, it doesn't do any I/O or networking, so it should be trivial to test.